diff --git a/bootable/recovery/data.cpp b/bootable/recovery/data.cpp
index 230ac73..a78d89b 100644
--- a/bootable/recovery/data.cpp
+++ b/bootable/recovery/data.cpp
@@ -1005,6 +1005,9 @@ int DataManager::GetMagicValue(const string varName, string& value)
 			char cap_s[4];
 #ifdef TW_CUSTOM_BATTERY_PATH
 			string capacity_file = EXPAND(TW_CUSTOM_BATTERY_PATH);
+#ifdef XPERIA_TWRP_TOUCH
+			capacity_file += "/ab8500_fg";
+#endif
 			capacity_file += "/capacity";
 			FILE * cap = fopen(capacity_file.c_str(),"rt");
 #else
@@ -1019,6 +1022,9 @@ int DataManager::GetMagicValue(const string varName, string& value)
 			}
 #ifdef TW_CUSTOM_BATTERY_PATH
 			string status_file = EXPAND(TW_CUSTOM_BATTERY_PATH);
+#ifdef XPERIA_TWRP_TOUCH
+			status_file += "/ab8500_chargalg";
+#endif
 			status_file += "/status";
 			cap = fopen(status_file.c_str(),"rt");
 #else
diff --git a/bootable/recovery/minuitwrp/events.c b/bootable/recovery/minuitwrp/events.c
index f07fc14..e88763e 100644
--- a/bootable/recovery/minuitwrp/events.c
+++ b/bootable/recovery/minuitwrp/events.c
@@ -168,7 +168,12 @@ static int vk_init(struct ev *e)
 #endif
 
     // Blacklist these "input" devices
+#ifdef XPERIA_TWRP_TOUCH
+    if (strcmp(e->deviceName, "bma250") == 0 || strcmp(e->deviceName, "bma150") == 0 
+        || strcmp(e->deviceName, "lsm303dlhc_acc_lt") == 0 || strcmp(e->deviceName, "lsm303dlh_mag") == 0)
+#else
     if (strcmp(e->deviceName, "bma250") == 0 || strcmp(e->deviceName, "bma150") == 0)
+#endif
     {
         e->ignored = 1;
     }
