diff --git a/bootable/recovery/minuitwrp/events.c b/bootable/recovery/minuitwrp/events.c
index f07fc14..e88763e 100644
--- a/bootable/recovery/minuitwrp/events.c
+++ b/bootable/recovery/minuitwrp/events.c
@@ -168,7 +168,12 @@ static int vk_init(struct ev *e)
 #endif
 
     // Blacklist these "input" devices
+#ifdef XPERIA_TWRP_TOUCH
+    if (strcmp(e->deviceName, "bma250") == 0 || strcmp(e->deviceName, "bma150") == 0 
+        || strcmp(e->deviceName, "lsm303dlhc_acc_lt") == 0 || strcmp(e->deviceName, "lsm303dlh_mag") == 0)
+#else
     if (strcmp(e->deviceName, "bma250") == 0 || strcmp(e->deviceName, "bma150") == 0)
+#endif
     {
         e->ignored = 1;
     }
